consumer/src/filters.rs

```diff
@@ -223,4 +272,133 @@ mod tests {
             common_filter(&tree.state().node_by_id(NodeId(1)).unwrap())
         );
     }
+
+    #[test]
+    fn clipped_children() {
```

Comment:
I feel like this test tests too many things at once, making it harder to understand if it fail in the future. Could you please try to split it so that each function contain less assertions?